### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by yfdyh000** [zluca/Sidebar#2]
1. "extDescription" to "Sidebar for Chromium and Firefox" or like "Sidebar panels".
1. No option to disable the startpage?
1. RSS Panel work?
1. Some options (input-number) and the search engine list is covered in 1366*768, and more serious if adjusted layout.css.devPixelsPerPx.
1. No context menu support. See also https://github.com/eoger/tabcenter-redux.
1. The bookmark panel is stuck when there are 10k bookmarks.

**Comment by zluca** [comment]
1. Agree 
2. Well, for now startpage hardcoded in the manifest and can not be disabled. I had experiments with injected startpage which can be disabled, but it works buggy in FF. But the main reason is that extension cannot inject iframe in startpage if it created by other extension or by default.
3. Works for me. What browser you use?
4. Accepted.
5. not sure about this. Anyway first priority is to fix bugs. I will think about it later.
6. Oo. I believe it because extension try to create fav icons for all of this bookmarks synchronically and it looks like "stuck". Need more experiments to check this theory. If you have time plz try to replace in "background.js" [this](https://github.com/zluca/Sidebar/blob/master/background.js#L1401) line with 
`newItem.domain  = 'default';`

**Comment by yfdyh000** [comment]
Item 3: Firefox Nightly. "Add new RSS feed" button looks no effect, throw `Error: Could not establish connection. Receiving end does not exist.`.
Well, I found the reason that it needs to load on the current page, you need to check the current page is not the privileged page (like about:addons) and use the notifications API to remind.

Item 6: Big improvement, just load takes a few seconds for the first start.

**Comment by yfdyh000** [comment]
7: A plan to adding Add-ons management panel?

**Comment by yfdyh000** [comment]
2: filled https://bugzilla.mozilla.org/show_bug.cgi?id=1420944

**Comment by zluca** [comment]
> Well, I found the reason that it needs to load on the current page, you need to check the current page is not the privileged page (like about:addons) and use the notifications API to remind.

It should open new tab if current page is privileged, i remember that it works before,  but now for some reason it broken. I will look into this. :/

> Item 6: Big improvement, just load takes a few seconds for the first start.

Ok, thx for testing. Now i have a clue.

> Allow users to control which one is enabled or all disabled for each chrome_url_overrides properties.

It would be nice if they implement this functionality.

> 7: A plan to adding Add-ons management panel?

No. ( If I correctly understand what you mean. =) )

**Comment by zluca** [comment]
@yfdyh000 
> 6. The bookmark panel is stuck when there are 10k bookmarks.

Can you tell me if it works better in latest version?

**Comment by yfdyh000** [comment]
> Can you tell me if it works better in latest version?

1. 0.3.4. It takes many seconds when enableing, high CPU usage.
2. If the bookmark panel is the last panel, it sometimes gets empty after it is enabled, `Error: Could not establish connection. Receiving end does not exist.` hundreds and the Search box's placeholder are `undefined`. Switch to another panel and switch back is work, switch back take about 2~5 seconds.
3. Perhaps it is better to provides max result options? Such as show recent bookmarks only by default, max search results, max folders and sub-items, `more...` item to folder filter.

**Comment by zluca** [comment]
>0.3.4.

plz test 0.3.5. ^^

>Error: Could not establish connection. Receiving end does not exist

Yep, annoying thing.

>show recent bookmarks only by default

Good idea, it can work! Tomorrow i will try.

**Comment by yfdyh000** [comment]
0.3.5 looks no changes. You can copy and paste several times to get a lot of bookmarks for testing.

**Comment by maverick74** [comment]
in Firefox I'm missing:

Synced Tabs
Pocket List
ScreenShoots

(in short, the items that are presented on **_Library_**)

Do any of this stand a chance???

**Comment by zluca** [comment]
@maverick74 
>Synced Tabs Pocket List

I need to make some research about this services. If they have good open API why not?

>ScreenShoots

But why you need sidebar for making screenshots? There is a tons of extensions providing this feature.

**Comment by maverick74** [comment]
about **ScreenShots**: i'm referring to the link/webpage that is open when one goes to Menu>Library>Screenshots https://screenshots.firefox.com/shots which contains one's screenshots, not the app to take screenshots.

**Pocket List** goes the same way.

In another words, **Pocket List** and **ScreenShots**, from what i understand, is just supposed to be a link/webpage that is open in the sidebar.

On the other hand Synced Tabs, i believe, is an app and not a webpage.

**Comment by zluca** [comment]
> is just supposed to be a link/webpage

That's how it looks from user view. =) Screenshots and Pocket have a build-in extensions which doing the job invisibly for user.

From what i see Pocket is the main candidate to implement in next big release. But no promises. :)

**Comment by zluca** [comment]
@yfdyh000 
> 2: filled https://bugzilla.mozilla.org/show_bug.cgi?id=1420944
> Priority: -- → P5

Ok, lets face the truth this is never happen. 😃
But users ask me so much about disabling startpage so i will do it.

**Comment by limon36** [comment]
yes because the all-in-one bar had it!!

**Comment by herbstgy** [comment]
is it possible to set the refresh interval of the RSS feeds?

**Comment by zluca** [comment]
@herbstgy 
Options > Misc > RSS update period

**Comment by herbstgy** [comment]
@zluca 
great, thx.

**Comment by EC-O-DE** [comment]
Very good! Just what I was looking for (well vertical bookmarks bar (hover in/hover out)) but this seems promising. Worked in Opera without much glitch.

There's two ext's that u dev might find inspirational:
Bookmarks Sidebar: https://chrome.google.com/webstore/detail/bookmark-sidebar/jdbnofccmhefkmjbkkdkfiicjkgofkdh?hl=en-US
And:
vTabs: https://chrome.google.com/webstore/detail/vtabs/okpnlgbgcfchbicbhjmmhldhkbkfilce?hl=en-US

**Comment by edger78** [comment]
Open bookmark in new tab is really useful feature. Could you please implement it?

**Comment by zluca** [comment]
@edger78 
You should hold Ctrl button to open bookmark in new tab, or Shift to open in new window.

**Comment by edger78** [comment]
Yes, I know that, thanks, wheel click also works this way. But simple click is very welcome option. Not only me using that. Not sure if it's too complex to implement it.

**Comment by zluca** [comment]
@edger78 
Ok now i got it. This option would be implemented, but  no ETAs, because i need to finish current development.

**Comment by edger78** [comment]
@zluca 
Thanks, sure. Take your time :)
It's really great to have it finally.

**Comment by herbstgy** [comment]
Hi,
is it possible to somehow delete the older items in the RSS feed? they just keep piling up ever since I installed the addon. (which I love by the way)
![image](https://user-images.githubusercontent.com/35459376/38139429-a7db8f90-342f-11e8-9765-cc55af0dfee7.png)

**Comment by zluca** [comment]
@herbstgy 
Older rss items should be deleted automatically,  there is an option in 'misc' section to set maximum items per feed. But for some reason it not work. Probably i broke something during last code refactor. Thx for reporting this bug.

**Comment by zluca** [comment]
@edger78 
Check out 0.5.5 version.

**Comment by edger78** [comment]
@zluca
Hi, checked, no such option found - open bookmark in new tab.
Default behavior the same.

**Comment by zluca** [comment]
@edger78 
Look to icon with 3 dots on the bottom of sidebar. :smiley:

**Comment by edger78** [comment]
@zluca
Wow, working like a charm!!
Thank you, really appreciate your work!

**Comment by zluca** [comment]
@herbstgy 
Fixed in 0.5.6. Sidebar+ clean old rss items at startup now.

**Comment by herbstgy** [comment]
@zluca 
neat, thanks.

**Comment by lrbloom** [comment]
Is there a way to move the sidebar to the left side of the browser?

**Comment by zluca** [comment]
@lrbloom 
Of course. You can have even to sidebars right and left. Click to the extension icon on the top of the browser.

**Comment by herbstgy** [comment]
this new 0.5.9 version doesn't work well.
i mean, it does not work at all. had to turn it off, because it prevents opening new pages.
is there a way to revert to 0.5.8?

**Comment by zluca** [comment]
@herbstgy 
[here](https://addons.mozilla.org/ru/firefox/addon/sidebar_plus/versions/) Better revert to 0.5.7. 
I disable 0.5.8 and 0.5.9 cause it works strange in Firefox.

**Comment by EC-O-DE** [comment]
![icons8-navigation-toolbar-left-96](https://user-images.githubusercontent.com/9984842/38764335-76b56972-3fb5-11e8-893a-9706ffe8278d.png)
![icons8-navigation-toolbar-left-64](https://user-images.githubusercontent.com/9984842/38764337-76d78f2a-3fb5-11e8-8b7c-ed5ac3bab909.png)
![gallery](https://user-images.githubusercontent.com/9984842/38764338-76fcd85c-3fb5-11e8-94fd-bf287c6830bf.png)
![if_64_sidebar_left_interface_menu_window_2142682](https://user-images.githubusercontent.com/9984842/38764339-7718c6f2-3fb5-11e8-8b2e-1a82fa535387.png)
![if_sidebar_left_interface_menu_window_1654369](https://user-images.githubusercontent.com/9984842/38764340-77454a6a-3fb5-11e8-9d76-0f702e37ffbc.png)


^^^ Few icons for logo/inspiration.
Btw: this is a rather good source: https://www.iconfinder.com/
Plenty of free ones.

**Comment by herbstgy** [comment]
now where have the autohide gone?
(FF 60.0.1, Sidebar+ 0.5.11)

**Comment by zluca** [comment]
@herbstgy 
Should work. Unless you check "Manual switch" in misc options or pin sidebar.

**Comment by herbstgy** [comment]
@zluca 
no, sorry.
I checked the option you mentioned, it is turned off. For some reason, autohide still not works. Anything else I should check?

**Comment by zluca** [comment]
@herbstgy 
Check if you use Iframe sidebar and it is not pinned.

**Comment by herbstgy** [comment]
I do use Iframe sidebar (tried to switch it back and forth a couple of times).
Where can I check the other? I do remember a pin icon from somewhere... but now i can't find it.

**Comment by herbstgy** [comment]
also, links don't open with a single click now... only with right click+open in new tab.
I'm thinking on an uninstall/reinstall magic now.

**Comment by herbstgy** [comment]
okay. the uninstall/reinstall magic worked.
now everything works again as it supposed to. (I've found the pin icon too :) )
but I lost my rss feeds in the process. :)
can you tell me here can i find the subscribed rss folder in the profiles?
I have backup, I can restore them from there.

**Comment by zluca** [comment]
@herbstgy 

By the way i suddenly remember this situation. If sidebar injected with old version of Sidebar+ and then update it loose connection with background script. So you have to update page.

**Comment by zluca** [comment]
> can you tell me here can i find the subscribed rss folder in the profiles?

Im afraid the is no folder. But you can export rss feeds to opml file.

**Comment by herbstgy** [comment]
too late, i lost them with the reinstall... :)

**Comment by kenreeser** [comment]
I would like the ability to open all bookmarks contained within a bookmark folder simultaneously via a newly added icon added to each lowest level bookmark folder

**Comment by zluca** [comment]
@kenreeser 
check out 0.5.16

**Comment by kenreeser** [comment]
Excellent. Thank you very much. I tried to locate a donation link, but was unsuccessful. Please forward one to me.

Since finally getting past the setup phase with Chrome, I've been able to spend quite a bit of time with it. I have some more questions and ideas about Sidebar+. Is there any way to have the Iframe views remain open? I realize I can use windows, but I've found them very difficult to use—they don't open in the same locations and with the dimensions in which last left them. In addition, every time I move Chrome to a different screen (I have 3) I have to drag 3 objects instead of 1. Also, is there way I can totally disable Sidebar+'s start page? I have another favorite start page extension that crossed over from Firefox, but Sidebar+ keeps overriding it. And lastly, just as helpful as now being able to open a folder full of bookmarks at one time, it would be as helpful to be able to hold down a modified key such Option (Alt on Windows) and click on any level bookmark folder and all bookmarks within that folder structure (all the way down) toggle between open and closed.

**Comment by zluca** [comment]
> I tried to locate a donation link, but was unsuccessful

Well, because there is none. I don't think it is worth it.
> Is there any way to have the Iframe views remain open

Yep, there is "Pin" button at the top of sidebar.
> Also, is there way I can totally disable Sidebar+'s start page?

Options > Services > Start page (uncheck it).
> it would be as helpful to be able to hold down a modified...

Maybe. =)

**Comment by kenreeser** [comment]
Well, I think it’s worth it. As a software developer myself, I’m a big believer in paying other developers for their time and ideas.

I tried unchecking the start page, but it continues to block the start page of my other extension.

I hope you're able to develop the folder-modifier-key toggle feature—just closing all my hierarchical bookmark folders was a real chore. ;-)

**Comment by kenreeser** [comment]
Is there any way to have Sidebar+ Iframes appear when my tab page extension (New Tab Tools) opens its a new tab page?

**Comment by zluca** [comment]
@kenreeser 
Unfortunately there is no way and that actually why i maked my own startpage. WebExtension are isolated from each other in a sandboxes.

**Comment by kenreeser** [comment]
Okay. So, my workaround (at least I thought) for being able to view my Sidebar+ menus (1 on either side of my browser window) upon Chrome loading was to send Chrome to specific page. Well that didn't work either. Upon loading chrome, it goes to the page I specified, but my sidebar+ menus are not visible—as if I started up on a tab page. Any ideas?

I can do everything I've been asking for to this point in Firefox. However, I've done a lot of performance/resource utilization tests between Firefox Quantum and Chrome and have found Chrome faster and uses less memory. I really want to switch to Chrome, but Google seems to be making it difficult.

**Comment by kenreeser** [comment]
Oh, by the way, the mouse configuration dialog is extremely tiny (on both Chrome and Quantum).

**Comment by zluca** [comment]
> I was wondering if Sidebar+'s Iframe menus could be made into split panes?

It's require a lot of work, so no. For now im just fixing bugs and do some little enhancements.

> I just attempted to move a single folder of bookmarks with Sidebar+ and now I have several hundred bookmark folders to close

Yep i'm agree hat this should be fixed, maybe some delay before folder opens.

**Comment by kenreeser** [comment]
How about only the hovered upon destination bookmark folder opens. And, if the a user changes their mind and wants to drag a bookmark item onto another destination, that prior destination bookmark folder closes and the next hovered upon destination opens. Actually, why can't this process work like the built-in bookmark managers in Firefox and Chrome?

**Comment by herbstgy** [comment]
I was enthusiastic at first but it simply don't work.
I gave up after a half dozen uninstall and reinstall.
Autohide don't work. i lost my RSS feeds and can't put them back.

**Comment by zluca** [comment]
@herbstgy 
I have a bunch of reports like this one. Problem is that I can't reproduce. It is maybe something related to browser/OS and I personally don't see because of using Firefox on Linux (that a rare setup).

**Comment by herbstgy** [comment]
@zluca 
thank you for you positive answer despite of my negative comment.
I didn't want to degrade your work, I know you put your heart in it.
The thing is, I've gotten kinda dependent of this plugin and disappoints me that I can't use it. :)
I tried several times to uninstall and reinstall it to clean up. Now I'm trying an older version (0.5.15), and I guess I'm on something... I get back to you later.

**Comment by herbstgy** [comment]
@zluca 
I think I got it.
Sidebar+ always works for a short period of time after I reinstall it... Until I start fiddling with the settings to set it to my liking. :)
the problem is with this dialog window:
![clipboard-1](https://user-images.githubusercontent.com/35459376/46684597-90b3e400-cbf3-11e8-8372-6fa138e60320.jpg)
I always try to set it up to open the favorites or the rss elements in a new tab in the browser. So, I set "single click" to "Open in new tab". ...And after that, the iFrame doesn't appear from the side anymore.

**Comment by herbstgy** [comment]
To add some info to the earlier: I'm on Win 10 64 bit and Firefox 62.0.3 64 bit.
also, the extension I use:
![image](https://user-images.githubusercontent.com/35459376/46684984-ac6bba00-cbf4-11e8-873e-7a0b97ddc3f6.png)

**Comment by zluca** [comment]
> I always try to set it up to open the favorites or the rss elements in a new tab in the browser. So, I set "Normal click" to "Open in new tab". ...And after that, the iFrame doesn't appear from the side anymore.

Thx, now I have something to start with.

**Comment by herbstgy** [comment]
I thought I have found the solution but after I restarted Firefox, autohide doesn't work again.
In iframe mode, if I move the mouse in the autohide activation area (the side of the window), nothing happens.  If I click the mouse in the autohide activation area, it initiates drag, resizing the Sidebar. I can't click on the links.
If I turn Sidebar+ in native mode, I can see the sidebar and the links, but clicks on the links don't work. I can only open them with right click + Open in new tab.

**Comment by zluca** [comment]
> I can see the sidebar and the links, but clicks on the links don't work. I can only open them with right click + Open in new tab.

This mean background script is dead. I fixed one possible reason in 9519e161, but there still is a lot job to do.

**Comment by herbstgy** [comment]
ever tried to use SideBar+ and Firefox in multiple windows on multiple monitors?
I move the mouse to the side for SideBar to appear in iframe mode in one window, and SideBar appears in the other window. :)

**Comment by zluca** [comment]
@herbstgy 
I guess you hover sidebar on inactive window and its appears in active, is it?

**Comment by herbstgy** [comment]
it's the opposite, really. :)

**Comment by zluca** [comment]
@herbstgy 
Is 0.6.0 release make things better?

**Comment by herbstgy** [comment]
now:
- if I open multiple Firefox widows, if I move the mouse to the side in the main (firstly opened) window, the sidebar appears in the right place.
- if I move the mouse to the side in the secondly opened window, nothing happens. sidebar does not appear in neither window.

- when I open a new instance of Firefox, an unwanted "New tab" appears beside the page I have opened, even if I turn off "Use custom startpage" option.

**Comment by zluca** [comment]
@herbstgy 
0.6.4 should fix iframe issues.

**Comment by yukon-cornelius** [comment]
Hello ... hope this suggestion list is still being monitored. :-)

Can a "reset settings" option please be added?  I was playing with colors, and once I changed the border color (which also changed each section's color headings?) ... I couldn't get back.  :-(

Also the settings are currently scattered around.  There's sorting at the bottom, options from FF's add-ons screen, mouse actions from a button-click at the bottom, and position & community & link to options in the toolbar.

Putting a background image either on the sidebar or the new start page neither says what file is being used, nor does the background imaqe state (enabled/disabled, background image itself) seem to update even after "Save" is clicked.

Lastly the "Cancel" button in Options doesn't seem to do anything.

Just a lot of fit & finish & thoughts about usability, that's all.  Thanks!

**Comment by zluca** [comment]
@yukon-cornelius 
Hello there. Active developing of this extension is frozen. Because the more experience i have the more clearly i understand that it was wrong architecture from the beginning. And i don't have enough courage (and time, and passion) to rewrite it from scratch as it meant to be. The only things i do now is small fixes.

> Can a "reset settings" option please be added?
Ok. 

> Putting a background image...
Accepted.

> Lastly the "Cancel" button in Options doesn't seem to do anything.
This button close floating "options dialog" window in Chromium based browsers. But in Firefox there is no dialog window so it is useless.

**Comment by maverick74** [comment]
Would be nice to have an option to "Open [page/link/bookmark] in sidebar" (Something like this: https://addons.mozilla.org/en-US/firefox/addon/open-link-in-sidebar/ ) - Just to avoid multiple extensions!
